lib/hip/hipProcess.cpp

```diff
@@ -0,0 +1,65 @@
+#include "hipProcess.hpp"
+
+#include "Config.hpp"
+#include "StageFactory.hpp"
+#include "factory.hpp"
+
+//#include "hip/hip_profiler_api.h"
+#include "unistd.h"
+#include "util.h"
+
+#include <iostream>
+#include <sys/time.h>
+
+using kotekan::bufferContainer;
+using kotekan::Config;
+
+using namespace std;
+
+REGISTER_KOTEKAN_STAGE(hipProcess);
+
+// TODO Remove the GPU_ID from this constructor
+hipProcess::hipProcess(Config& config_, const string& unique_name,
+                       bufferContainer& buffer_container) :
+    gpuProcess(config_, unique_name, buffer_container) {
+    device = new hipDeviceInterface(config_, gpu_id, _gpu_buffer_depth);
+    dev = device;
+    device->prepareStreams();
+    // CHECK_HIP_ERROR(hipProfilerStart());
```

Comment:
This should probably be a config option rather than a hard-uncomment, but I don't think worth holding up the merge on account of.

Reply:
We already have a `profiling` configuration option in the base class, so that's an easy change.  However now, I'm not sure why this was commented out in the first place.  I was running profiling tests on this branch, so either this is a commit error, or we don't actually need this line.   I'll check this before merging.

Reply:
Turns out this deals with the external profiling application, not the internal timers.  So it's simply not needed, will remove.